geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/geotools/SimpleFeatureTypes.scala

```diff
@@ -143,10 +144,19 @@ object SimpleFeatureTypes {
     renamed
   }
 
-  def encodeType(sft: SimpleFeatureType, includeUserData: Boolean = false): String = {
-    val suffix = if (!includeUserData || sft.getUserData.isEmpty) { "" } else {
-      sft.getUserData.entrySet.map(e => s"${e.getKey}='${e.getValue}'").mkString(";", ",", "")
+  def encodeType(sft: SimpleFeatureType): String = {
```

Comment:
hmm, i don't think we would want to encode all the user data when e.g. passing an sft to an iterator or something. Would it work if we just persisted the geomesa user data when we write the sft to the metadata? that would also allow us to get rid of the duplicative meta-data entries.